### ISSUE TITLE
m3u url panics when channel ID is missing

### ISSUE THREAD

**Issue by blockloop** [euzu/m3u-filter#61]
**Describe the bug**
When an m3u playlist doesn't have a channel ID a panic is raised [here](https://github.com/euzu/m3u-filter/blob/f0092e27c728103c30342ffe445cd5c040223f8f/src/repository/xtream_repository.rs#L257)

**To Reproduce**
1. Load playlist containing channels with `tvg-id=""`
2. See panic

```
iptv-m3u-filter-1  | [2024-05-25T04:01:45Z INFO  actix_server::server] Actix runtime found; starting in Actix runtime
iptv-m3u-filter-1  | thread 'main' panicked at src/repository/xtream_repository.rs:257:95:
iptv-m3u-filter-1  | already mutably borrowed: BorrowError
iptv-m3u-filter-1  | stack backtrace:
iptv-m3u-filter-1  |    0:     0x7f79411e4ba1 - std::backtrace_rs::backtrace::libunwind::trace::h103e47e00552e141
iptv-m3u-filter-1  |                                at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/../../backtrace/src/backtrace/libunwind.rs:105:5
iptv-m3u-filter-1  |    1:     0x7f79411e4ba1 - std::backtrace_rs::backtrace::trace_unsynchronized::h592221613f1fd095
iptv-m3u-filter-1  |                                at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
iptv-m3u-filter-1  |    2:     0x7f79411e4ba1 - std::sys_common::backtrace::_print_fmt::hf8a30f8299b1bef3
iptv-m3u-filter-1  |                                at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/sys_common/backtrace.rs:68:5
iptv-m3u-filter-1  |    3:     0x7f79411e4ba1 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h5a6e3e7cc140e753
iptv-m3u-filter-1  |                                at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/sys_common/backtrace.rs:44:22
iptv-m3u-filter-1  |    4:     0x7f794101b43c - core::fmt::rt::Argument::fmt::hd62a89729b8d4079
iptv-m3u-filter-1  |                                at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/core/src/fmt/rt.rs:142:9
iptv-m3u-filter-1  |    5:     0x7f794101b43c - core::fmt::write::haa8a32127a7b0d75
iptv-m3u-filter-1  |                                at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/core/src/fmt/mod.rs:1153:17
iptv-m3u-filter-1  |    6:     0x7f79411bd09e - std::io::Write::write_fmt::hc541adb06d0aeac1
iptv-m3u-filter-1  |                                at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/io/mod.rs:1843:15
iptv-m3u-filter-1  |    7:     0x7f79411e9c4f - std::sys_common::backtrace::_print::h59dc0d5369ede572
iptv-m3u-filter-1  |                                at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/sys_common/backtrace.rs:47:5
iptv-m3u-filter-1  |    8:     0x7f79411e9c4f - std::sys_common::backtrace::print::hbb7b999054d983a7
iptv-m3u-filter-1  |                                at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/sys_common/backtrace.rs:34:9
iptv-m3u-filter-1  |    9:     0x7f79411e955d - std::panicking::default_hook::{{closure}}::hd143a60ad2fc5a07
iptv-m3u-filter-1  |   10:     0x7f79411ea470 - std::panicking::default_hook::h2aff2dd57e84e09f
iptv-m3u-filter-1  |                                at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/panicking.rs:292:9
iptv-m3u-filter-1  |   11:     0x7f79411ea470 - std::panicking::rust_panic_with_hook::h6811313e71f57da8
iptv-m3u-filter-1  |                                at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/panicking.rs:779:13
iptv-m3u-filter-1  |   12:     0x7f79411e9fb0 - std::panicking::begin_panic_handler::{{closure}}::h2eb4991eec554504
iptv-m3u-filter-1  |                                at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/panicking.rs:657:13
iptv-m3u-filter-1  |   13:     0x7f79411e9f06 - std::sys_common::backtrace::__rust_end_short_backtrace::h05983ae26e2c8878
iptv-m3u-filter-1  |                                at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/sys_common/backtrace.rs:171:18
iptv-m3u-filter-1  |   14:     0x7f79411e9ef3 - rust_begin_unwind
iptv-m3u-filter-1  |                                at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/panicking.rs:645:5
iptv-m3u-filter-1  |   15:     0x7f7940e79074 - core::panicking::panic_fmt::h49eed7d6e9368ad2
iptv-m3u-filter-1  |                                at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/core/src/panicking.rs:72:14
iptv-m3u-filter-1  |   16:     0x7f7940e79408 - core::cell::panic_already_mutably_borrowed::hcadc424dc75a6e9d
iptv-m3u-filter-1  |                                at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/core/src/cell.rs:778:5
iptv-m3u-filter-1  |   17:     0x7f7940f9e7ba - core::cell::RefCell<T>::borrow::h0583f79262590506
iptv-m3u-filter-1  |   18:     0x7f7940f46346 - m3u_filter::repository::xtream_repository::xtream_write_playlist::h4f014e8fc47284bd
iptv-m3u-filter-1  |   19:     0x7f7940f61777 - m3u_filter::processing::playlist_processor::process_playlist::{{closure}}::hac1685d5d0e9fed6
iptv-m3u-filter-1  |   20:     0x7f7940f5b645 - m3u_filter::processing::playlist_processor::process_source::{{closure}}::h94217eee680d13a0
iptv-m3u-filter-1  |   21:     0x7f7940f65f70 - m3u_filter::processing::playlist_processor::exec_processing::{{closure}}::hdf6b1d154131ce1c
iptv-m3u-filter-1  |   22:     0x7f7940ee156d - tokio::runtime::task::raw::poll::h1b85bec42757be42
iptv-m3u-filter-1  |   23:     0x7f79411f2cc9 - tokio::task::local::LocalSet::tick::h9177afd50173e071
iptv-m3u-filter-1  |   24:     0x7f7940ee03f9 - tokio::task::local::LocalSet::run_until::{{closure}}::h15f104b6cbb93897
iptv-m3u-filter-1  |   25:     0x7f7940f7ce47 - m3u_filter::main::hb7eae3e9f4c84f8b
iptv-m3u-filter-1  |   26:     0x7f7940ec305c - std::sys_common::backtrace::__rust_begin_short_backtrace::h10fb5c891061533a
iptv-m3u-filter-1  |   27:     0x7f7940f86920 - main
```

**Comment by euzu** [comment]
@blockloop i have pushed the changes to the master, please give me a feedback after testing

**Comment by euzu** [comment]
closing due inactivity, should be solved with v2.0.2
